### PR TITLE
Add `nsimplify` call in case limit argument has a `Float`

### LINKED ISSUE #12312: Sympy returns an unevaluated limit
Trying 
```
>>> e = 2**(-x)*(sin(x) + 1)**x
>>> limit(e, x, oo)
Limit(2**(-x)*(sin(x) + 1)**x, x, oo, dir='-')
```
returns an unevaluated limit.

### LINKED ISSUE #19453: Limit changes from simplification of original expression
I have an ugly symbolic expression:
`E = -(-0.375*hbar**3*beta*g*exp(0.5*hbar*beta*omega)/(m**2*omega*(exp(hbar*beta*omega) - 1)) - 0.75*hbar**3*beta*g*exp(0.5*hbar*beta*omega)/(m**2*omega*(exp(hbar*beta*omega) - 1)**2) + 3*hbar**3*beta*g*exp(1.5*hbar*beta*omega)/(4*m**2*omega*(exp(hbar*beta*omega) - 1)**2) - 0.75*hbar**3*beta*g*(exp(hbar*beta*omega) + 1)*exp(0.5*hbar*beta*omega)/(m**2*omega*(exp(hbar*beta*omega) - 1)**3) + 3*hbar**3*beta*g*exp(1.5*hbar*beta*omega)/(2*m**2*omega*(exp(hbar*beta*omega) - 1)**3) + 9*hbar**3*beta*g*(exp(hbar*beta*omega) + 1)*exp(1.5*hbar*beta*omega)/(2*m**2*omega*(exp(hbar*beta*omega) - 1)**4) - 3*hbar**2*g*exp(0.5*hbar*beta*omega)/(4*m**2*omega**2*(exp(hbar*beta*omega) - 1)) - 3*hbar**2*g*exp(0.5*hbar*beta*omega)/(2*m**2*omega**2*(exp(hbar*beta*omega) - 1)**2) - 3*hbar**2*g*(exp(hbar*beta*omega) + 1)*exp(0.5*hbar*beta*omega)/(2*m**2*omega**2*(exp(hbar*beta*omega) - 1)**3) + 0.5*hbar*omega*exp(0.5*hbar*beta*omega)/(exp(hbar*beta*omega) - 1) - hbar*omega*exp(1.5*hbar*beta*omega)/(exp(hbar*beta*omega) - 1)**2)/(-3*hbar**2*beta*g*exp(0.5*hbar*beta*omega)/(4*m**2*omega**2*(exp(hbar*beta*omega) - 1)) - 3*hbar**2*beta*g*exp(0.5*hbar*beta*omega)/(2*m**2*omega**2*(exp(hbar*beta*omega) - 1)**2) - 3*hbar**2*beta*g*(exp(hbar*beta*omega) + 1)*exp(0.5*hbar*beta*omega)/(2*m**2*omega**2*(exp(hbar*beta*omega) - 1)**3) + exp(0.5*hbar*beta*omega)/(exp(hbar*beta*omega) - 1))
`
If I call:
>`limit(E.simplify()-0.5*hbar*w, beta, oo)`

The result is 0
But if I call 
>`limit(E-0.5*hbar*w, beta, oo)`

The result is claimed to be infinity.
I think the two expressions should be equivalent

### PR DESCRIPTION
**Pull request by 0sidharth**
#### References to other Issues or PRs

Fixes #12312
Fixes #19453 

#### Brief description of what is fixed or changed
Convert floats like `0.5` to exact SymPy numbers like `S.Half`, to
prevent rounding errors which can lead to unexpected execution
of conditional blocks that work on comparisons

#### Other comments
See #19453 

#### Release Notes

* series
  * Limit now converts `Float`'s in the expression to `Rational`s or exact SymPy numbers when possible

### CODE REVIEW

File: sympy/series/tests/test_limits.py [hunk 1]
```diff
@@ -156,7 +156,7 @@ def test_floor():
 
     # https://github.com/sympy/sympy/issues/14478
     assert limit(x*floor(3/x)/2, x, 0, '+') == Rational(3, 2)
-    assert limit(floor(x + 1/2) - floor(x), x, oo) == AccumBounds(-0.5, 1.5)
+    assert limit(floor(x + 1/2) - floor(x), x, oo) == AccumBounds(-S.Half, S(3)/2)
```
Comment (oscargus) [inline sympy/series/tests/test_limits.py:159]: I think this should really have been `S(1)/2` or `S.Half` to start with. Nothing in the issue seems to indicate that this was an issue.

Comment (oscargus) [reply sympy/series/tests/test_limits.py:159]: I know it is not related to the thing you changed though, but still...

Comment (0sidharth) [reply sympy/series/tests/test_limits.py:159]: The limit currently matches with [wolfram alpha](https://www.wolframalpha.com/input/?i=limit%28floor%28x+%2B+1%2F2%29+-+floor%28x%29%2C+x%2C+oo%29), though I am not sure why this result is so... I would have thought it should be `AccumBounds(0, 1)`.

Comment (0sidharth) [reply sympy/series/tests/test_limits.py:159]: After going through the comments on the PR, this limit seems to be the cause of the other linked issue (#14313) (Ref this [comment](https://github.com/sympy/sympy/pull/20803#discussion_r560066567))

**Comment by jksuom** [comment]
Looks good.

**Pull request merged**